targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W2/sdk/ublox-odin-w2-drivers/OdinWiFiInterface.cpp

```diff
@@ -230,6 +242,29 @@ nsapi_error_t OdinWiFiInterface::set_credentials(const char *ssid, const char *p
     return NSAPI_ERROR_OK;
 }
 
+nsapi_error_t OdinWiFiInterface::set_certificates(const char *client_cert, const char *client_pvt_key , const char *ca_cert )
+{
+    int size;
+
+    /* Validate size of certificates passed, max allowed size is 4096 */
+    VALIDATE_CERT(client_cert);
+    VALIDATE_CERT(client_pvt_key);
+    VALIDATE_CERT(ca_cert);
+
+    /* Check if client certificate is valid */
+    if((client_cert != NULL) && (*client_cert != '\0')) {
+        _certs.client_cert = client_cert;
+        _certs.client_prvt_key = client_pvt_key;
```

Comment:
Isn't the check for `client_pvt_key` missing here?
And shouldn't the NULL pointer check precede the VALIDATE_CERT macro calls? `strlen()` will segfault on NULL pointer...